runtime/lib/mirrors_impl.dart

```diff
@@ -1665,7 +1667,10 @@ class _Mirrors {
     return classMirror;
   }
 
-  static TypeMirror reflectType(Type key) {
+  static TypeMirror reflectType(Type key, [List<Type> typeArguments]) {
+    if (typeArguments != null) {
+      key = _resolveType(key, typeArguments);
```

Comment:
Fine with me.
I wanted to suggest something similar, but in the end it looked more difficult than necessary. Maybe I missed something.